integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/trace/TracingInterceptor.kt

```diff
@@ -788,16 +789,23 @@ internal constructor(
             return getThis()
         }
 
+        /**
+         * Set the trace sample rate controlling the sampling of APM traces created for
+         * auto-instrumented requests.
+         * @param sampleRate the sample rate to use (percentage between 0f and 100f, default is 20f).
+         */
+        fun setTraceSampleRate(@FloatRange(from = 0.0, to = 100.0) sampleRate: Float): R {
+            this.traceSampler = RateBasedSampler(sampleRate)
+            return getThis()
+        }
+
         /**
          * Set the trace sampler controlling the sampling of APM traces created for
          * auto-instrumented requests.
          * @param traceSampler the trace sampler controlling the sampling of APM traces.
-         * By default it is [RateBasedSampler], which either can accept
-         * fixed sample rate or can get it dynamically from the provider. Value between `0.0` and
-         * `100.0`. A value of `0.0` means no trace will be kept, `100.0` means all traces will
-         * be kept (default value is `20.0`).
+         * By default it is a sampler accepting 20% of the traces.
          */
-        fun setTraceSampler(traceSampler: Sampler): R {
+        fun setTraceSampler(traceSampler: Sampler<Span>): R {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏ Like I suggested in my previous comment maybe we can keep also the previous method and wrap the customer old `Sampler` implementation in our own `Sampler<Span>` which delegates to the customer. Not sure but I think is doable. We can also `Deprecate` the old method.